File: Models/Entities/Report.cs
using System.ComponentModel.DataAnnotations;

namespace szakdoga.Models
{
    public class Report : BaseEntity
    {
        [Required]
        [StringLength(50)]
        public string ReportGUID { get; set; }

        public Query Query { get; set; }

        /// <summary>
        /// A grafikonok adataihoz
        /// </summary>
        public string Style { get; set; }

        /// <summary>
        /// :-vel tagolt string "tömb"
        /// </summary>
        public string Columns { get; set; }

        public string Filter { get; set; }
        public string Sort { get; set; }
        public int Rows { get; set; }
    }
}

File: ReportingWebApplication/ClientApp/app/components/report/report.ts
import { IEntityWithIdName, IColumnSort } from "../shared/shared-interfaces";

export interface IReportList {
    totalCount: number;
    reports: IReport[];
}

export interface IReport {
    reportGUID: string;
    name: string;
    query: IEntityWithIdName;
    creationDate: Date;
    author: IEntityWithIdName;
    modifyDate: Date;
    lastModifier: IEntityWithIdName;
    hasStyle: boolean;
}

export interface IReportCreate {
    name: string;
    queryGUID: string;
    columns: string[];
    filter: string;
    rows: number;
    sort: IColumnSort;
    //Style: string;
}

//export class Report implements IReport {
//    constructor(public id: number,
//        public name: string,
//        public query: IEntityWithIdName,
//        public createdAt: string,
//        public createdBy: IEntityWithIdName,
//        public modifiedAt: string,
//        public modifiedBy: IEntityWithIdName,
//        public style: string) {
//    }

//    // műveletek
//}